dataplicity/portforward.py

```diff
@@ -24,12 +24,19 @@ class Connection(threading.Thread):
     # Max to read at-a-time
     BUFFER_SIZE = 1024 * 32
 
-    def __init__(self, service, connection_id, channel):
+    def __init__(self, close_event, connection_id, channel,
+                 service=None, host_port=None):
         """Initialize the connection, set up callbacks."""
         super(Connection, self).__init__()
-        self._service = weakref.ref(service)
+        self._close_event = close_event
         self.connection_id = connection_id
         self.channel = channel
+        self._service = None
+        # this is for backward compatibility purposes
```

Comment:
What *backwards compatibility*?

Reply:
as I re-read the code, I must have meant the `service` attribute itself. The Connection thread is used for both named services and the dynamic port redirects, which don't have any service. I think that naming it backwards compatibility may have been unfortunate. Would you agree that I remove it ?